src/test/java/au/com/centrumsystems/hudson/plugin/buildpipeline/trigger/BuildPipelineTriggerTest.java

```diff
@@ -247,6 +248,87 @@ public void testManualTriggerCause() throws Exception
     }
 
     @Test
+    @Issue("JENKINS-23532")
+    public void testTriggerProjectInFolderUsingParameteriezedTrigger()
+        throws Exception
+    {
+        //root folder
+        MockFolder folder1 = jenkins.createFolder("Folder1");
+        MockFolder folder2 = jenkins.createFolder("Folder2");
+
+        //  /Folder1
+        final String projUp = "upstream";
+        folder1.createProject(FreeStyleProject.class, projUp);
+        //FreeStyleProject upstreamProject = (FreeStyleProject) folder1.getItem(projUp);
+        FreeStyleProject upstreamProject = jenkins.createFreeStyleProject(projUp);
+
+        //  /Folder2
+        final String projDown = "downstream";
+        folder2.createProject(FreeStyleProject.class, projDown);
+        FreeStyleProject downstreamProject = (FreeStyleProject) folder2.getItem(projDown);
+
+        BuildTriggerConfig notFakeConfig = new BuildTriggerConfig("Folder2/downstream", ResultCondition.ALWAYS, false,
+                Collections.singletonList((AbstractBuildParameters) new CurrentBuildParameters()));
+        upstreamProject.getPublishersList().add(new hudson.plugins.parameterizedtrigger.BuildTrigger(notFakeConfig));
+        jenkins.getInstance().rebuildDependencyGraph();
+
+        BuildPipelineView view = new BuildPipelineView("Pipeline", "Title", new DownstreamProjectGridBuilder("upstream"), "1", false, "");
+        view.setBuildCard(new StandardBuildCard());
+        jenkins.buildAndAssertSuccess(upstreamProject);
+
+        view.triggerManualBuild(1, downstreamProject.getFullName(), upstreamProject.getFullName());

```

Comment:
Has upstream automatically started to build the downstream? And if so, does this mean that we are triggering a re-run and not a first-run, which had the problem we must reproduce? You can verify this by asserting null on the downstream project's latest build. If it has in fact started, a work-around would be to define the downstream project and configuration in the upstream project after the upstream has been run once.